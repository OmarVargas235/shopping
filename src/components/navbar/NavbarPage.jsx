import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { NavLink } from 'react-router-dom';

import { ReactComponent  as ShoppingCart } from '../../assets/iconos/shoppingCart.svg';

const Nav = styled.nav`
	.navbar__link {
		color: black;
		margin-right: 3rem;
	}

	.navbar__link--active {
		text-decoration: underline;
	}

	.navbar__link--mr {
		margin-right: 0;
	}
`;

const NavbarPage = ({ products }) => (
	<Nav className="d-flex justify-content-center py-3 mb-4">
		<NavLink to='/home' className="navbar__link" activeClassName="navbar__link--active">Store</NavLink>
		<NavLink to='/about' className="navbar__link" activeClassName="navbar__link--active">About</NavLink>
		<NavLink to='/cart' className="navbar__link navbar__link--mr d-flex" activeClassName="navbar__link--active">
			<span className="mr-1 d-flex align-items-center"> <ShoppingCart /> </span> Cart ({products.length})
		</NavLink>
	</Nav>
)

NavbarPage.propTypes = {
	products: PropTypes.array, 
}

NavbarPage.defaultProps = {
	products: [], 
}

export default NavbarPage;