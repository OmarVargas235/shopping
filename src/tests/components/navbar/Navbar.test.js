import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import toJson from 'enzyme-to-json';
import NavbarPage from '../../../components/navbar/NavbarPage';

test('Pruebas en <NavbarPage />', () => {	
	const wrapper = shallow(
		<NavbarPage 
			products={[1,2,3,4]}
		/>
	);
	
	expect(toJson(wrapper)).toMatchSnapshot();
});