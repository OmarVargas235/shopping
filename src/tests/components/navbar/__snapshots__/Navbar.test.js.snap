// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <NavbarPage /> 1`] = `
<styled.nav
  className="d-flex justify-content-center py-3 mb-4"
>
  <NavLink
    activeClassName="navbar__link--active"
    className="navbar__link"
    to="/home"
  >
    Store
  </NavLink>
  <NavLink
    activeClassName="navbar__link--active"
    className="navbar__link"
    to="/about"
  >
    About
  </NavLink>
  <NavLink
    activeClassName="navbar__link--active"
    className="navbar__link navbar__link--mr d-flex"
    to="/cart"
  >
    <span
      className="mr-1 d-flex align-items-center"
    >
       
      <ForwardRef(SvgShoppingCart) />
       
    </span>
     Cart (
    4
    )
  </NavLink>
</styled.nav>
`;
